js/piwik.js

```diff
@@ -7071,6 +7074,15 @@ if (typeof window.Matomo !== 'object') {
             });
 
             Matomo.trigger('TrackerSetup', [this]);
+
+            Matomo.addPlugin('TrackerVisitorIdCookie' + uniqueTrackerId, {
+                // if no tracking request was sent we refresh the visitor id cookie on page unload
+                unload: function () {
+                    if (!hasSentTrackingRequestYet) {
+                        setVisitorIdCookie();
```

Comment:
Wouldn't this mean, that if the user leaves the page (without previous tracking), but did not yet gave consent a cookie would be set nevertheless?

Reply:
Also remember previously we called `setVisitorIdCookie()` much earlier, so this doesn't change behaviour other than delaying this call.

Reply:
was just wondering as you do a check for `(!configCookiesDisabled && !getCookie(getCookieName('id')))` within `getVisitorInfo`, but not here.

Reply:
good point, I'll actually remove the check on `getVisitorInfo()` as it is unnecessary.